dotcom-rendering/src/components/Footer.tsx

```diff
@@ -241,15 +243,26 @@ const FooterLinks = ({
 	const linkGroups = pageFooter.footerLinks.map((linkGroup) => {
 		const linkList = linkGroup.map(
 			({ url, extraClasses, dataLinkName, text }) => (
-				<li key={`${url}`}>
-					<a
-						css={[footerLink, extraClasses]}
-						href={url}
-						data-link-name={dataLinkName}
-					>
-						{text}
-					</a>
-				</li>
+				<Fragment key={url}>
+					{dataLinkName === 'privacy' && (
+						<li key="client-side-island">
+							<Island>
+								<PrivacySettingsLink
+									extraClasses={extraClasses}
+								/>
+							</Island>
+						</li>
+					)}
```

Comment:
Is there a better pattern to inject the privacy settings link? Should we enhance the `footerLinks`?

Reply:
it must be better to only hydrate the thing that's changing in a situation like this? we know the rest will never change, so what do we get? would it promote dev ergonomics at the cost of unnecessary CPU cycles?

Reply:
By “enhance” I meant modify the array as a preprocessing step, not actually hydrate the whole list! I’m just wondering wether this current pattern of wrapping things in a fragment is confusing.